src/wallet/spend.cpp

```diff
@@ -557,14 +559,24 @@ std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, cons
         results.push_back(*srd_result);
     }
 
-    if (results.size() == 0) {
+    if (results.empty()) {
         // No solution found
         return std::nullopt;
     }
```

Comment:
As `std::copy_if` is a no-op if the vector is empty, could remove these lines.

Reply:
Leaving the `results.empty())` check would also avoid creating the `eligible_results` no?

Reply:
yes, would only avoid the empty vector initialization.